### DISCUSSION TITLE
Dividing files into smaller modules

### DISCUSSION THREAD

**Discussion by jeevithakannan2** [ChrisTitusTech/linutil#844] [Ideas]
Should we split the files into small modes for modularity and maintainability ?

For example seperating the draw function and handle_function from state.rs into separate files as ui.rs and keybindings.rs and having a parent module as app state 

Summoning the gods here @ChrisTitusTech @lj3954 @adamperkowski @cartercanedy @nnyyxxxx @JustLinuxUser @afonsofrancof

**Comment by koibtw** [comment]
Yea I think that's a good idea. I always split things like that in my projects.

**Comment by cartercanedy** [comment]
Totally, I'd just wait until we start paring down that list of 73 PRs

**Comment by jeevithakannan2** [comment]
Yeah I will start working on splitting `state.rs`.

app_state.rs ( App state focus enum, list entry struct and new fn)

hints.rs ( keybinding hints )
keybindings.rs ( handle_keys )

misc.rs ( Functions like selected_item_is_cmd, enable_preview, refresh_tab, etc )
mod.rs ( module )
ui.rs ( draw function )
tips.rs ( cool tips )

This what I was thinking about, is this overkill ?? or is it okay ??

**Reply by ghost** [reply]
unless it needs to be separated its best we keep it together

**Reply by ghost** [reply]
separating code can seem good but then you'll end up not knowing what goes to what afterwards

**Reply by ghost** [reply]
if we were to separate state.rs its best we put the separated code into its own subdir to quell the matter above

**Reply by jeevithakannan2** [reply]
> if we were to separate state.rs its best we put the separated code into its own subdir to quell the matter above

Yeah that's what a parent module means. All these files will be in `state` dir. And mod state is used in the main file to source the module.

**Reply by jeevithakannan2** [reply]
> unless it needs to be separated its best we keep it together

The code base of state.rs is too big and in future it will become harder to maintain it.


> separating code can seem good but then you'll end up not knowing what goes to what afterwards

From the above suggestion does it make it worse ?

**Reply by cartercanedy** [reply]
Yeah, @nnyyxxxx has a point. I don't really want to get to a point that this becomes like every other C# project with only a single struct/impl per file, that's the absolute worst imho

**Reply by cartercanedy** [reply]
That's not to say that you shouldn't do this, but let's not go hog wild with it

**Reply by jeevithakannan2** [reply]
If you have suggestions they are welcome. That's why I am here

**Reply by jeevithakannan2** [reply]
I will merge Focus and ListEntry with AppState. Other files have their core functionality which cannot be merged with other files. Maybe hints and keybindings can be merged. If you guys have any suggestions let me know !!

**Reply by koibtw** [reply]
> Yeah, @nnyyxxxx has a point. I don't really want to get to point that this become like every other C# project with only a single struct/impl per file, that's the absolute worst imho
>
> That's not to say that you shouldn't do this, but let's not go hog wild with it

Yeah I kinda agree
